src/components/Form.js

```diff
@@ -74,42 +105,46 @@ export default class Form extends Component {
     return null;
   }
 
-  onChange = (formData, options={validate: false}) => {
+  onChange = (formData, options = {validate: false}) => {
     const mustValidate = !this.props.noValidate && (this.props.liveValidate || options.validate);
-    let state = {status: "editing", formData};
-    if (mustValidate) {
-      const {errors, errorSchema} = this.validate(formData);
-      state = {...state, errors, errorSchema};
-    }
-    setState(this, state, () => {
-      if (this.props.onChange) {
-        this.props.onChange(this.state);
-      }
+    const onChange = this.props.onChange;
+
+    this.setState(Object.assign({}, this.state, {status: "editing", formData}), () => {
+      Promise.resolve(mustValidate ? this.validate(formData) : {})
+        .then(this.setValidationResult.bind(this))
```

Comment:
Be very careful as setting state within a promise flow may throw and call the global catch event. See this thread on twitter https://twitter.com/dan_abramov/status/798856890813845504 (basically you have to provide a catch function along `setValidationResult` here)